### PR TITLE
Mech Equipment Carepackage 1: Mainly Weapon Slots

### PR DESCRIPTION
**Pull request by Mechoid**
First of probably 3 PRs for more Mecha components. The next will be hull slots primarily, which will involve more of a rework on how they process armor and the like.

Features: 
- Boatload of new exosuit weapons, and one new hull item.
- Most exosuit weapons now use accuracy traits from their pilot. Inaccurate pilots may have trouble hitting the broad-side of the outpost.

Tweaks:
- The flashbang launcher is now a generalized 'grenade launcher', which uses the proper grenade activation method. Untouched otherwise.
- Whole bunch of sprites added, rigged laser given its own special sprite to match the new rigged theme.
- Exosuit modules can now do things on-move.

RESEARCH PRODUCEABLE:
Standard:

- Concussion grenade launcher; Fires a new type of grenade, a 'concussion' grenade, which is really just a flashbang without the flash. It does not work in vacuum due to no blast medium.
- Mini-frag launcher; Fires tiny frag grenades with small blast radii, fewer pieces of shrapnel, and weaker shrapnel in general.
- Rapid-fire but high-power-draw Xray laser; Standard x-ray laser, like the carbine orderable from cargo.
- Exomounted phase gun; Rapid fire, relatively low charge. Fits in utility as a 'side-arm' for combat mechs or a primary defense tool for Ripleys down in the caves.
- Shield generator; Creates a linear combat shield in the direction the exosuit is facing when activated. Draws power constantly while active, recharging, or taking damage. Will cut itself off if the power is too low.

Jury-Rigged:
- Redone welding laser; Is actually comparable to a normal laser, but less efficient and slightly slower.
- X-ray gun; less efficient, twice as slow.
- Emitter Cannon; Mines rock walls like a standard emitter, is equivalent to a solaris in damage, but has a far longer recharge time and larger drain.
- Ion gun; Fires ion pistol projectiles, is actually faster and slightly less charge-hungry than the normal one, because of its tiny projectile.
- Taser; uses the weak taser beam used by the FM-2t, slightly more energy hungry.
- Shotgun; awful deviation, half-sized magazine, and a 1 second higher cooldown due to being a massive sawnoff, essentially.
- LMG; Uses a weaker bullet, has a higher spread, and slightly higher cooldown.
- Rocket pod; Uses a weaker missile with no heavy damage, and a smaller magazine. Energy drain is slightly decreased from the main variant.
- Flashbang launcher; magazine of 3, slower grenade flight time, and a longer grenade priming time.

NON-RESEARCH PRODUCEABLE
- True frag launcher; Fires Actual Frag Grenades, with a magazine of 4.
- Mortar Tube; Fires arcing mortar shells, smaller than the standard fragmentation projectile. Can't fire anywhere indoors.

### CODE REVIEW

File: code/game/mecha/equipment/mecha_equipment.dm [hunk 1]
```diff
@@ -19,6 +19,7 @@
 	var/salvageable = 1
 	var/required_type = /obj/mecha //may be either a type or a list of allowed types
 	var/equip_type = null //mechaequip2
+	var/allow_duplicate = 0
```
Comment (Anewbe) [inline code/game/mecha/equipment/mecha_equipment.dm]: Could this not be FALSE/TRUE?

File: code/game/mecha/equipment/mecha_equipment.dm [hunk 2]
```diff
@@ -103,15 +104,19 @@
 		return 0
 	return 1
 
+/obj/item/mecha_parts/mecha_equipment/proc/handle_movement_action() //Any modules that have special effects or needs when moving.
```
Comment (Anewbe) [inline code/game/mecha/equipment/mecha_equipment.dm]: Can you explain in this comment what you mean by moving?

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 3]
```diff
@@ -12,6 +12,7 @@
 	var/inject_amount = 5
 	required_type = /obj/mecha/medical
 	salvageable = 0
+	allow_duplicate = 1
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/medical_tools.dm]: See the above about FALSE/TRUE.

File: code/game/mecha/equipment/tools/tools.dm [hunk 4]
```diff
@@ -1072,6 +1153,7 @@
 	var/mob/living/carbon/occupant = null
 	var/door_locked = 1
 	salvageable = 0
+	allow_duplicate = 1
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: FALSE/TRUE

File: code/game/mecha/equipment/tools/tools.dm [hunk 5]
```diff
@@ -827,6 +826,88 @@
...
+	my_shield.my_tool = null
+	my_shield.my_mecha = null
+	qdel(my_shield)
+	my_shield = null
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/combat_shield/attach(obj/mecha/M as obj)
+	..()
+	if(chassis)
+		my_shield.shield_health = my_shield.shield_health / 2
```
Comment (Anewbe) [inline code/game/mecha/equipment/tools/tools.dm]: What specifically does this line do?

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 6]
```diff
@@ -47,56 +47,157 @@
...
+	origin_tech = list(TECH_COMBAT = 4, TECH_MAGNET = 4, TECH_PHORON = 3, TECH_ILLEGAL = 1)
+
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/phase
+	equip_cooldown = 6
+	name = "\improper NT-PE \"Scorpio\" phase-emitter"
+	desc = "A specialist energy weapon intended for use against hostile wildlife."
+	description_fluff = "Essentially an Orion mounted inside a modified Gaia case."
+	icon_state = "mecha_phase"
+	energy_drain = 25
+	projectile = /obj/item/projectile/energy/phase
```
Comment (Anewbe) [inline code/game/mecha/equipment/weapons/weapons.dm]: If this is supposed to be an Orion, give it the `phase/heavy` type.

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 7]
```diff
@@ -256,17 +416,40 @@
 	icon_state = "missile"
 	var/primed = null
 	throwforce = 15
+	var/devastation = 0
+	var/heavy_blast = 1
+	var/light_blast = 2
+	var/flash_blast = 4
+
+/obj/item/missile/proc/warhead_special(var/target)
+	explosion(target, devastation, heavy_blast, light_blast, flash_blast)
```
Comment (Anewbe) [inline code/game/mecha/equipment/weapons/weapons.dm]: Does this proc accept atoms? Thought it was only turfs?

File: code/game/mecha/equipment/tools/tools.dm (hunk 5 above)
Comment (Mechoid) [reply code/game/mecha/equipment/tools/tools.dm]: Sets the shield to half health when the shield is attached, as a 'cheese' prevention, and as a windup. I realized writing this that it's equally cheezable.

File: code/game/mecha/equipment/weapons/weapons.dm (hunk 7 above)
Comment (Mechoid) [reply code/game/mecha/equipment/weapons/weapons.dm]: Warhead special for the default is a normal explosion, another type if made might accept like, doors, or something, as a special breaching explosion, or a human as a netmissile gimmick.

Comment (Anewbe) [reply code/game/mecha/equipment/weapons/weapons.dm]: I mean the `explosion` proc itself.

Comment (Mechoid) [reply code/game/mecha/equipment/weapons/weapons.dm]: The missiles did in fact explode when I tested them, so it appears so. I'll re-test.

Comment (Mechoid) [reply code/game/mecha/equipment/weapons/weapons.dm]: Yep, the explosion is made on the target atom.

**Pull request merged**